File: Задание3/prog3.cs
using System;

public class Program
{
    public static int NumReverse(int num)
    {
        int length = 0;
        int numCopy = num;
        int digit = 10;
        
        while (numCopy != 0)
        {
            if (numCopy % 10 >= digit)
            {
                return 0;
            }
            
            length += 1;
            digit = numCopy % 10;
            numCopy /= 10;
        }
        
        for (int i = 0; i < length; i++)
        {
            numCopy = numCopy * 10 + num % 10;
            num /= 10;
        }
        
        return length > 1 ? numCopy : 0;
    }
    
    public static void Main()
    {
        int count = int.Parse(Console.ReadLine());
        
        while (count > 0)
        {
            int number = int.Parse(Console.ReadLine());
            int result = NumReverse(number);
            
            if (result != 0)
            {
                Console.Write(result + " ");
            }
            
            count--;
        }
    }
}

File: Задание3/prog3.go
package main

import (
	"fmt"
	"strconv"
)

func numReverse(num int) int {
	length := 0
	numCopy := num
	digit := 10

	for numCopy > 0 {
		if numCopy%10 >= digit {
			return 0
		}

		length++
		digit = numCopy % 10
		numCopy /= 10
	}

	numCopy = 0
	for i := 0; i < length; i++ {
		numCopy = numCopy*10 + num%10
		num /= 10
	}

	if length > 1 {
		return numCopy
	} else {
		return 0
	}
}

func main() {
	var count int
	fmt.Scan(&count)

	for count > 0 {
		var number int
		fmt.Scan(&number)
		result := numReverse(number)

		if result != 0 {
			fmt.Print(strconv.Itoa(result) + " ")
		}

		count--
	}
}